extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java

```diff
@@ -0,0 +1,172 @@
+        private String consumes;
+
+        public MethodDescription(String method, String fullPath, String produces, String consumes) {
+            super();
+            this.method = method;
+            this.fullPath = fullPath;
+            this.produces = produces;
+            this.consumes = consumes;
+        }
+
+    }
+
+    private static final class ResourceDescription {
+        private String basePath;
+        private List<MethodDescription> calls;
+
+        public ResourceDescription(String basePath) {
+            this.basePath = basePath;
+            this.calls = new ArrayList<>();
+        }
+
+        public void addMethod(String path, ResourceMethodInvoker method) {
+            String produces = mostPreferredOrNull(method.getProduces());
+            String consumes = mostPreferredOrNull(method.getConsumes());
+
+            for (String verb : method.getHttpMethods()) {
+                calls.add(new MethodDescription(verb, path, produces, consumes));
+            }
+        }
+
+        private static String mostPreferredOrNull(MediaType[] mediaTypes) {
+            if (mediaTypes == null || mediaTypes.length < 1) {
+                return null;
+            } else {
+                return mediaTypes[0].toString();
+            }
+        }
+
+        public static List<ResourceDescription> fromBoundResourceInvokers(
+                Set<Map.Entry<String, List<ResourceInvoker>>> bound) {
+            Map<String, ResourceDescription> descriptions = new HashMap<>();
+
+            for (Map.Entry<String, List<ResourceInvoker>> entry : bound) {
+                Method aMethod = ((ResourceMethodInvoker) entry.getValue().get(0)).getMethod();
+                String basePath = aMethod.getDeclaringClass().getAnnotation(Path.class).value();
+
+                if (!descriptions.containsKey(basePath)) {
+                    descriptions.put(basePath, new ResourceDescription(basePath));
+                }
+
+                for (ResourceInvoker invoker : entry.getValue()) {
+                    ResourceMethodInvoker method = (ResourceMethodInvoker) invoker;
+
+                    String subPath = null;
+                    for (Annotation annotation : method.getMethodAnnotations()) {
+                        if (annotation.annotationType().equals(Path.class)) {
+                            subPath = ((Path) annotation).value();
+                            break;
+                        }
+                    }
+
+                    descriptions.get(basePath).addMethod(basePath + subPath, method);
+                }
+            }
+
+            return new LinkedList<>(descriptions.values());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Response toResponse(NotFoundException exception) {
+        TemplateHtmlBuilder sb = new TemplateHtmlBuilder();
+        if (registry == null || !LaunchMode.DEVELOPMENT.getDefaultProfile().equals(ProfileManager.getActiveProfile())) {
```

Comment:
I think you can leave the test to `registry == null` and skip the part about profiles, since you've already made sure this exception mapper is only registered in DEV mode.